tools/fetch-node/src/main/java/org/sonarsource/javascript/XZ.java

```
@@ -37,25 +40,41 @@ public static void main(String[] args) throws Exception {
     compress(args);
   }
 
-  public static void compress(String[] filenames) throws Exception {
+  /**
+   * Compress the provided filenames with the `DEFAULT_COMPRESSION_LEVEL`
+   *
+   * @param filenames
+   * @throws Exception
+   */
+  public static void compress(String[] filenames) throws IOException {
+    compress(filenames, DEFAULT_COMPRESSION_LEVEL);
+  }
+
+  /**
+   * Compress with a custom compression level, used for tests
+   *
+   * @param filenames
+   * @param compressionLevel
+   * @throws Exception
+   */
+  public static void compress(String[] filenames, int compressionLevel) throws IOException {
     for (var filename : filenames) {
       System.out.println("Compressing " + filename);
       var file = Path.of(filename);
       if (!Files.exists(file)) {
-        System.out.println("File " + filename + " does not exist.");
-        System.exit(1);
+        throw new FileNotFoundException("File " + filename + " does not exist.");
       }
       try (
         var is = Files.newInputStream(file);
         var outfile = Files.newOutputStream(Path.of(file + ".xz"));
-        var outxz = new XZOutputStream(outfile, new LZMA2Options(COMPRESSION_LEVEL))
+        var outxz = new XZOutputStream(outfile, new LZMA2Options(compressionLevel))
       ) {
         is.transferTo(outxz);
       }
       try {
         Files.delete(file);
-      } catch (Exception e) {
-        System.out.println("Error while deleting file: " + e.getMessage());
+      } catch (IOException e) {
```

no need to catch the exception, just let it bubble up

---

done